packages/dds/sequence/src/sequence.ts

```diff
@@ -397,8 +398,9 @@ export abstract class SharedSegmentSequence<T extends MergeTree.ISegment>
     public async waitIntervalCollection(
         label: string,
     ): Promise<IntervalCollection<SequenceInterval>> {
-        return this.intervalMapKernel.wait<IntervalCollection<SequenceInterval>>(
-            this.getIntervalCollectionPath(label));
+        return this.intervalMapKernel.wait(
+            this.getIntervalCollectionPath(label) as unknown as Serializable,
+        ) as unknown as IntervalCollection<SequenceInterval>;
     }
```

Comment:
Wow.  Applying the new Serializable to Map correctly flagged this sketchy internal code that is inserting a class with prototype & functions into the map kernel.

Later it uses `intervalCollection.attached` to detect whether the member function should be accessible.  Since a detached collection can only exist on the client that created it, we get away with it.

Reply:
Is this not normal valuetype usage?  I'm less familiar with Sequence's usage of it, but value types would be the exception where it appears a legit object can be retrieved from a mapKernel.

Reply:
Woah...  Didn't realize value types were still around.  Looks like it's only exposed on MapKernel?

Thanks for pointing out my error.  I'm still going to count this example as a win, since Serializable<T> is correct to flag this as non-serializable.  :-)  ...but will definitely fix the typing on MapKernel when doing the SharedMap conversion "for real".

When you have a chance, have a look at #6316, which is the part I'm hoping to land first.

Reply:
Sounds good.  FWIW, I think we should split MapKernel and move the value type handling out of the map package and into Sequence.